### ISSUE TITLE
React Router v5.1.1 Link Breaks with React v15.6.2

### ISSUE THREAD

**Issue by macpham** [ReactTraining/react-router#6954]
## Version

v5.1.1

## Steps to reproduce

1. Using `react` `v15.6.2` and `react-router` `v5.1.1`
2. **Production build (minified) with webpack v2 and Uglify** (not sure if pertinent to the issue but part of the setup I am using)
3. Create a component that renders with:

```
<div>
  <Link to={`/watch=${someVar}`}>Direct Link{</Link>
</div>
```

## Expected Behavior

No errors.

## Actual Behavior

Console will show:

![image](https://user-images.githubusercontent.com/1777724/65884167-58f2fe80-e366-11e9-991a-d5d5542b0384.png)

https://reactjs.org/docs/error-decoder.html/?invariant=119

Note, the problem doesn't seem to happen when we are doing a development build where nothing is minified. I am not sure exactly why that is the case. Our production build is targeting ES5.

This problem does not happen at all when we are on `v5.0.1` which we have been using for months. As a work around we have pinned our RR back to this version but could you please look into this as we would like to stay in sync with your latest releases.

**Comment by StringEpsilon** [comment]
Could this be a follow-up error of the #6934 fix? 

The 'minified react error \#119' says:

```
addComponentAsRefTo(...): Only a ReactOwner can have refs. You might be adding a ref to a component that was not created inside a component's `render` method, or you have multiple copies of React loaded (details: https://fb.me/react-refs-must-have-owner).
```

https://reactjs.org/docs/error-decoder.html/?invariant=119

That smells pretty much like another ref error.

**Comment by mjackson** [comment]
This should be fixed in 5.1.2, released soon
